src/utils/useInterval.ts

```diff
@@ -0,0 +1,27 @@
+import { useRef, useEffect, MutableRefObject } from 'react';
+
+/**
+ * React hook for setting up and using intervals.
+ *
+ * @param callback - Function to be called every interval tick
+ * @param delay - Delay in ms between interval ticks
+ * @param runImmediately - Whether the callback should be run at the begging
+ */
+export const useInterval = (callback: Function, delay: number, runImmediately: boolean = false) => {
```

Comment:
I can see there is no `return` for `useInterval` hook, is timer properly cleared, when `onClick` (`<Button variant="primary" onClick={refreshBuildCounts}>`) is executed?

Reply:
I believe interval should be properly cleaned even without returning anything. The useEffect should take care of everything.

Reply:
The useEffect is clearing it only when component is unmounted